File: js_sdk/src/model/CommentTaskCreateBody.js
/*
 * Copyright 2023 TikTok Pte. Ltd.
 *
 * This source code is licensed under the MIT license found in
 * the LICENSE file in the root directory of this source tree.
 */
import {ApiClient} from '../ApiClient.js';

/**
 * The CommentTaskCreateBody model module.
 * @module model/CommentTaskCreateBody
 * @version 0.1.7
 */
export class CommentTaskCreateBody {
  /**
   * Constructs a new <code>CommentTaskCreateBody</code>.
   * @alias module:model/CommentTaskCreateBody
   * @class
   * @param advertiser_id {String} 
   * @param end_time {String} 
   * @param search_field {String} 
   * @param search_value {String} 
   * @param start_time {String} 
   */
  constructor(advertiser_id, end_time, search_field, search_value, start_time) {
    this.advertiser_id = advertiser_id;
    this.end_time = end_time;
    this.search_field = search_field;
    this.search_value = search_value;
    this.start_time = start_time;
  }

  /**
   * Constructs a <code>CommentTaskCreateBody</code> from a plain JavaScript object, optionally creating a new instance.
   * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
   * @param {Object} data The plain JavaScript object bearing properties of interest.
   * @param {module:model/CommentTaskCreateBody} obj Optional instance to populate.
   * @return {module:model/CommentTaskCreateBody} The populated <code>CommentTaskCreateBody</code> instance.
   */
  static constructFromObject(data, obj) {
    if (data) {
      obj = obj || new CommentTaskCreateBody();
      if (data.hasOwnProperty('advertiser_id'))
        obj.advertiser_id = ApiClient.convertToType(data['advertiser_id'], 'String');
      if (data.hasOwnProperty('comment_ids'))
        obj.comment_ids = ApiClient.convertToType(data['comment_ids'], ['String']);
      if (data.hasOwnProperty('comment_status'))
        obj.comment_status = ApiClient.convertToType(data['comment_status'], ['String']);
      if (data.hasOwnProperty('comment_type'))
        obj.comment_type = ApiClient.convertToType(data['comment_type'], ['String']);
      if (data.hasOwnProperty('end_time'))
        obj.end_time = ApiClient.convertToType(data['end_time'], 'String');
      if (data.hasOwnProperty('lang'))
        obj.lang = ApiClient.convertToType(data['lang'], 'String');
      if (data.hasOwnProperty('search_field'))
        obj.search_field = ApiClient.convertToType(data['search_field'], 'String');
      if (data.hasOwnProperty('search_value'))
        obj.search_value = ApiClient.convertToType(data['search_value'], 'String');
      if (data.hasOwnProperty('sort_field'))
        obj.sort_field = ApiClient.convertToType(data['sort_field'], 'String');
      if (data.hasOwnProperty('sort_type'))
        obj.sort_type = ApiClient.convertToType(data['sort_type'], 'String');
      if (data.hasOwnProperty('start_time'))
        obj.start_time = ApiClient.convertToType(data['start_time'], 'String');
    }
    return obj;
  }
}

/**
 * @member {String} advertiser_id
 */
CommentTaskCreateBody.prototype.advertiser_id = undefined;

/**
 * @member {Array.<String>} comment_ids
 */
CommentTaskCreateBody.prototype.comment_ids = undefined;

/**
 * @member {Array.<String>} comment_status
 */
CommentTaskCreateBody.prototype.comment_status = undefined;

/**
 * @member {Array.<String>} comment_type
 */
CommentTaskCreateBody.prototype.comment_type = undefined;

/**
 * @member {String} end_time
 */
CommentTaskCreateBody.prototype.end_time = undefined;

/**
 * @member {String} lang
 * @default 'EN'
 */
CommentTaskCreateBody.prototype.lang = 'EN';

/**
 * @member {String} search_field
 */
CommentTaskCreateBody.prototype.search_field = undefined;

/**
 * @member {String} search_value
 */
CommentTaskCreateBody.prototype.search_value = undefined;

/**
 * @member {String} sort_field
 * @default 'CREATE_TIME'
 */
CommentTaskCreateBody.prototype.sort_field = 'CREATE_TIME';

/**
 * @member {String} sort_type
 * @default 'DESC'
 */
CommentTaskCreateBody.prototype.sort_type = 'DESC';

/**
 * @member {String} start_time
 */
CommentTaskCreateBody.prototype.start_time = undefined;

File: java_sdk/src/main/java/business_api_client/CommentTaskCreateBody.java
/*
 * Copyright 2023 TikTok Pte. Ltd.
 *
 * This source code is licensed under the MIT license found in
 * the LICENSE file in the root directory of this source tree.
 */

package business_api_client;

import java.util.Objects;
import java.util.Arrays;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonValue;
import io.swagger.v3.oas.annotations.media.Schema;
import java.util.ArrayList;
import java.util.List;
/**
 * CommentTaskCreateBody
 */

@javax.annotation.Generated(value = "io.swagger.codegen.v3.generators.java.JavaClientCodegen", date = "2026-01-08T15:25:30.628143-08:00[America/Los_Angeles]")
public class CommentTaskCreateBody {
  @JsonProperty("advertiser_id")
  private String advertiserId = null;

  @JsonProperty("comment_ids")
  private List<String> commentIds = null;

  @JsonProperty("comment_status")
  private List<String> commentStatus = null;

  @JsonProperty("comment_type")
  private List<String> commentType = null;

  @JsonProperty("end_time")
  private String endTime = null;

  @JsonProperty("lang")
  private String lang = "EN";

  @JsonProperty("search_field")
  private String searchField = null;

  @JsonProperty("search_value")
  private String searchValue = null;

  @JsonProperty("sort_field")
  private String sortField = "CREATE_TIME";

  @JsonProperty("sort_type")
  private String sortType = "DESC";

  @JsonProperty("start_time")
  private String startTime = null;

  public CommentTaskCreateBody advertiserId(String advertiserId) {
    this.advertiserId = advertiserId;
    return this;
  }

   /**
   * Get advertiserId
   * @return advertiserId
  **/
  @Schema(required = true, description = "")
  public String getAdvertiserId() {
    return advertiserId;
  }

  public void setAdvertiserId(String advertiserId) {
    this.advertiserId = advertiserId;
  }

  public CommentTaskCreateBody commentIds(List<String> commentIds) {
    this.commentIds = commentIds;
    return this;
  }

  public CommentTaskCreateBody addCommentIdsItem(String commentIdsItem) {
    if (this.commentIds == null) {
      this.commentIds = new ArrayList<String>();
    }
    this.commentIds.add(commentIdsItem);
    return this;
  }

   /**
   * Get commentIds
   * @return commentIds
  **/
  @Schema(description = "")
  public List<String> getCommentIds() {
    return commentIds;
  }

  public void setCommentIds(List<String> commentIds) {
    this.commentIds = commentIds;
  }

  public CommentTaskCreateBody commentStatus(List<String> commentStatus) {
    this.commentStatus = commentStatus;
    return this;
  }

  public CommentTaskCreateBody addCommentStatusItem(String commentStatusItem) {
    if (this.commentStatus == null) {
      this.commentStatus = new ArrayList<String>();
    }
    this.commentStatus.add(commentStatusItem);
    return this;
  }

   /**
   * Get commentStatus
   * @return commentStatus
  **/
  @Schema(description = "")
  public List<String> getCommentStatus() {
    return commentStatus;
  }

  public void setCommentStatus(List<String> commentStatus) {
    this.commentStatus = commentStatus;
  }

  public CommentTaskCreateBody commentType(List<String> commentType) {
    this.commentType = commentType;
    return this;
  }

  public CommentTaskCreateBody addCommentTypeItem(String commentTypeItem) {
    if (this.commentType == null) {
      this.commentType = new ArrayList<String>();
    }
    this.commentType.add(commentTypeItem);
    return this;
  }

   /**
   * Get commentType
   * @return commentType
  **/
  @Schema(description = "")
  public List<String> getCommentType() {
    return commentType;
  }

  public void setCommentType(List<String> commentType) {
    this.commentType = commentType;
  }

  public CommentTaskCreateBody endTime(String endTime) {
    this.endTime = endTime;
    return this;
  }

   /**
   * Get endTime
   * @return endTime
  **/
  @Schema(required = true, description = "")
  public String getEndTime() {
    return endTime;
  }

  public void setEndTime(String endTime) {
    this.endTime = endTime;
  }

  public CommentTaskCreateBody lang(String lang) {
    this.lang = lang;
    return this;
  }

   /**
   * Get lang
   * @return lang
  **/
  @Schema(description = "")
  public String getLang() {
    return lang;
  }

  public void setLang(String lang) {
    this.lang = lang;
  }

  public CommentTaskCreateBody searchField(String searchField) {
    this.searchField = searchField;
    return this;
  }

   /**
   * Get searchField
   * @return searchField
  **/
  @Schema(required = true, description = "")
  public String getSearchField() {
    return searchField;
  }

  public void setSearchField(String searchField) {
    this.searchField = searchField;
  }

  public CommentTaskCreateBody searchValue(String searchValue) {
    this.searchValue = searchValue;
    return this;
  }

   /**
   * Get searchValue
   * @return searchValue
  **/
  @Schema(required = true, description = "")
  public String getSearchValue() {
    return searchValue;
  }

  public void setSearchValue(String searchValue) {
    this.searchValue = searchValue;
  }

  public CommentTaskCreateBody sortField(String sortField) {
    this.sortField = sortField;
    return this;
  }

   /**
   * Get sortField
   * @return sortField
  **/
  @Schema(description = "")
  public String getSortField() {
    return sortField;
  }

  public void setSortField(String sortField) {
    this.sortField = sortField;
  }

  public CommentTaskCreateBody sortType(String sortType) {
    this.sortType = sortType;
    return this;
  }

   /**
   * Get sortType
   * @return sortType
  **/
  @Schema(description = "")
  public String getSortType() {
    return sortType;
  }

  public void setSortType(String sortType) {
    this.sortType = sortType;
  }

  public CommentTaskCreateBody startTime(String startTime) {
    this.startTime = startTime;
    return this;
  }

   /**
   * Get startTime
   * @return startTime
  **/
  @Schema(required = true, description = "")
  public String getStartTime() {
    return startTime;
  }

  public void setStartTime(String startTime) {
    this.startTime = startTime;
  }


  @Override
  public boolean equals(java.lang.Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    CommentTaskCreateBody commentTaskCreateBody = (CommentTaskCreateBody) o;
    return Objects.equals(this.advertiserId, commentTaskCreateBody.advertiserId) &&
        Objects.equals(this.commentIds, commentTaskCreateBody.commentIds) &&
        Objects.equals(this.commentStatus, commentTaskCreateBody.commentStatus) &&
        Objects.equals(this.commentType, commentTaskCreateBody.commentType) &&
        Objects.equals(this.endTime, commentTaskCreateBody.endTime) &&
        Objects.equals(this.lang, commentTaskCreateBody.lang) &&
        Objects.equals(this.searchField, commentTaskCreateBody.searchField) &&
        Objects.equals(this.searchValue, commentTaskCreateBody.searchValue) &&
        Objects.equals(this.sortField, commentTaskCreateBody.sortField) &&
        Objects.equals(this.sortType, commentTaskCreateBody.sortType) &&
        Objects.equals(this.startTime, commentTaskCreateBody.startTime);
  }

  @Override
  public int hashCode() {
    return Objects.hash(advertiserId, commentIds, commentStatus, commentType, endTime, lang, searchField, searchValue, sortField, sortType, startTime);
  }


  @Override
  public String toString() {
    StringBuilder sb = new StringBuilder();
    sb.append("class CommentTaskCreateBody {\n");
    
    sb.append("    advertiserId: ").append(toIndentedString(advertiserId)).append("\n");
    sb.append("    commentIds: ").append(toIndentedString(commentIds)).append("\n");
    sb.append("    commentStatus: ").append(toIndentedString(commentStatus)).append("\n");
    sb.append("    commentType: ").append(toIndentedString(commentType)).append("\n");
    sb.append("    endTime: ").append(toIndentedString(endTime)).append("\n");
    sb.append("    lang: ").append(toIndentedString(lang)).append("\n");
    sb.append("    searchField: ").append(toIndentedString(searchField)).append("\n");
    sb.append("    searchValue: ").append(toIndentedString(searchValue)).append("\n");
    sb.append("    sortField: ").append(toIndentedString(sortField)).append("\n");
    sb.append("    sortType: ").append(toIndentedString(sortType)).append("\n");
    sb.append("    startTime: ").append(toIndentedString(startTime)).append("\n");
    sb.append("}");
    return sb.toString();
  }

  /**
   * Convert the given object to string with each line indented by 4 spaces
   * (except the first line).
   */
  private String toIndentedString(java.lang.Object o) {
    if (o == null) {
      return "null";
    }
    return o.toString().replace("\n", "\n    ");
  }

}
